consent/helpers.py

```diff
@@ -0,0 +1,162 @@
+# -*- coding: utf-8 -*-
+"""
+Helper functions for the Consent application.
+"""
+
+from __future__ import absolute_import, unicode_literals
+
+from django.contrib.auth.models import User
+
+# Will be using internal models at a later time.
+from enterprise.models import (
+    EnterpriseCourseEnrollment,
+    EnterpriseCustomer,
+    EnterpriseCustomerUser,
+    UserDataSharingConsentAudit,
+)
+
+
+def consent_exists(user_id, course_id, enterprise_customer_uuid):
+    """
+    Get whether any consent is associated with an EnterpriseCustomer.
+
+    :param user_id: The user that grants consent.
+    :param course_id: The course for which consent is granted.
+    :param enterprise_customer_uuid: The consent requester.
+    :return: Whether any consent (provided or unprovided) exists.
+    """
+    enterprise_customer = get_enterprise_customer(enterprise_customer_uuid)
+    enterprise_course_enrollment = get_enterprise_course_enrollment(user_id, course_id, enterprise_customer)
+
+    if enterprise_course_enrollment is None or not enterprise_course_enrollment.consent_available:
+        return (enterprise_course_enrollment is not None or
+                get_user_dsc_audit(user_id, enterprise_customer) is not None)
+
+    return True
+
+
+def consent_provided(user_id, course_id, enterprise_customer_uuid):
+    """
+    Get whether consent is provided by the user to the Enterprise customer.
+
+    :param user_id: The user that grants consent.
+    :param course_id: The course for which consent is granted.
+    :param enterprise_customer_uuid: The consent requester.
+    :return: Whether consent is provided to the Enterprise customer by the user for a course.
+    """
+    enterprise_customer = get_enterprise_customer(enterprise_customer_uuid)
+    enterprise_course_enrollment = get_enterprise_course_enrollment(user_id, course_id, enterprise_customer)
+    provided = enterprise_course_enrollment is not None and enterprise_course_enrollment.consent_available
+
+    if not provided or enterprise_course_enrollment is None:
+        user_data_consent_audit = get_user_dsc_audit(user_id, enterprise_customer)
+        provided = user_data_consent_audit is not None and user_data_consent_audit.enabled
+
+    return provided is not None and provided is not False
+
+
+def consent_required(user_id, course_id, enterprise_customer_uuid):
```

Comment:
Yeah at first I had a pretty bad understanding of what the relation was between consent being required and consent being provided. Can you double check the changes I made to some of the existing test's `consent_required` expectations to see if they make sense?